crates/biome_js_analyze/src/analyzers/nursery/use_regex_literals.rs

```diff
@@ -0,0 +1,251 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, FixKind, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::js_regex_literal_expression;
+use biome_js_semantic::SemanticModel;
+use biome_js_syntax::{
+    global_identifier, AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression,
+    AnyJsTemplateElement, JsCallArguments, JsCallExpression, JsComputedMemberExpression,
+    JsNewExpression, JsSyntaxKind, JsSyntaxToken,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, AstNodeList, AstSeparatedList, BatchMutationExt, SyntaxError,
+};
+
+use crate::{semantic_services::Semantic, JsRuleAction};
+
+declare_rule! {
+    /// Enforce the use of the regular expression literals instead of the RegExp constructor if possible.
+    ///
+    /// It is considered a best practice to avoid the string literal notation on top of the regular expression notation,
+    /// and use regular expression literals instead of the constructor function.
+    ///
```

Comment:
I suggest adding more context and why it is better to use regex literals:

```suggestion
    /// There are two ways to create a regular expression:
    /// - Regular expression literals, e.g., `/abc/u`.
   /// - The RegExp constructor function, e.g., `new RegExp("abc", "u")` .
   ///
   /// The constructor function is particularly useful when you want to dynamically generate the pattern,
   /// because it takes string arguments.
   ///
   /// Using regular expression literals avoids some escaping required in a string literal,
   /// and are easier to analyze statically.
   ///
```